webapp/channels/src/components/channel_header/channel_header_title_favorite.tsx

```diff
@@ -19,17 +19,20 @@ const ChannelHeaderTitleFavorite = () => {
     const dispatch = useDispatch();
     const isFavorite = useSelector(isCurrentChannelFavorite);
     const channel = useSelector(getCurrentChannel);
-    const channelIsArchived = channel.delete_at !== 0;
+    const channelIsArchived = channel?.delete_at !== 0;
```

Comment:
This will return `true` if the channel is `undefined`

Reply:
I find this a "nice side-effect". A channel that is not in the store is, at least for the client, the same as a channel that doesn't exist, and that can be considered an archived channel. What do you think?

(The behavior change here is that instead of crashing the channel shows as archived, so... an improvement still :P )

Reply:
Personally I'm not a huge fan of side-effect programming, because that context around archived channels isn't immediately apparent to someone reading the code for the first time, so it's a bit misleading.

Also, archived does not always mean not exist (eg. users can view archived channels with the one config item set). If the channel doesn't exist, does that mean we had an error fetching it or is this a check in case we have a channel ID that might have been deleted? 

The intention here is important - we were actually talking about how to handle these in the Web Guild meeting.

To me, the "correct" solution would be to say `(channel?.delete_at ?? 0) > 0` which would return `false` if the channel was undefined. Or maybe it should return `undefined` since we don't have a channel to check for. 
```suggestion
    const channelIsArchived = (channel?.delete_at ?? 0) > 0;
```